pkg/kubelet/kubelet_node_status.go

```diff
@@ -538,16 +537,23 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *v1.Node) {
 		node.Status.NodeInfo.MachineID = info.MachineID
 		node.Status.NodeInfo.SystemUUID = info.SystemUUID
 
-		for rName, rCap := range cadvisor.CapacityFromMachineInfo(info) {
-			node.Status.Capacity[rName] = rCap
-		}
-
-		if kl.podsPerCore > 0 {
-			node.Status.Capacity[v1.ResourcePods] = *resource.NewQuantity(
-				int64(math.Min(float64(info.NumCores*kl.podsPerCore), float64(kl.maxPods))), resource.DecimalSI)
-		} else {
-			node.Status.Capacity[v1.ResourcePods] = *resource.NewQuantity(
-				int64(kl.maxPods), resource.DecimalSI)
+		node.Status.Capacity = kl.containerManager.GetCapacity()
```

Comment:
I'm suspecting there is an issue with changing this line.  The old behavior was calling the cadvisor interface to get machine info every time node status is called.  This PR changes it to only get machineinfo during kubelet initialization.  And maybe there is some ordering issue where cadvisor is not initialized yet during kubelet init.

Still not sure why it is an issue in the kubemark test and not the e2e test.